src/core/cardano/walletConnect/peerConnection.ts

```diff
@@ -30,7 +30,7 @@ class PeerConnection {
   private identityWalletConnect: IdentityWalletConnect | undefined;
   private connected = false;
 
-  async start() {
+  async start(selectedAid: string) {
```

Comment:
Right now in the UI if we connect to 2 meerkat instances, it will call `start` twice which just overrides the meerkat instance.

For now, if we are already connected with a dApp, we should call disconnect before connecting the new dApp in `start`. So a simple check at the start of this function to call disconnect if needed should be enough.

I'll get Jorge to make his document signing app made available so you can test this with something real.